MHArmory.Search/OpenCL/Host.cs

```diff
@@ -0,0 +1,109 @@
+using System;
+using System.Collections.ObjectModel;
+using System.Text;
+using Cloo;
+
+namespace MHArmory.Search.OpenCL
+{
+    class Host : IDisposable
+    {
+        private ComputeContext Context { get; }
+        private ComputeProgram Program { get; }
+        private const string KernelName = "search";
+
+        public Host()
+        {
+            byte[] source = TrimUTFBOM(Properties.Resources.search);
+            string sourceStr = Encoding.ASCII.GetString(source);
+
+            // TODO: We need a proper platform / device selection mechanism.
+            ComputePlatform platform = ComputePlatform.Platforms[2];
```

Comment:
I think it is better to have 0 set by default. I assume this is your default working value but I also assume in most cases people will have 1 device, 2 devices at best, making your value to crash in 99% of cases.

Reply:
This will later be replaced by a selector in the UI providing the index, probably to the `Host` class constructor.

Reply:
I'll add a proper way to do this by an `IComputeDeviceResolver`, I'll give it a default behavior of auto-picking the most performant GPU on the system. Then it will be easy to pass values from the GUI with a different implementation of the interface, which reads the GUI or something like that.